cluster.tf

```diff
@@ -1,12 +1,12 @@
 resource "digitalocean_kubernetes_cluster" "application-cluster" {
-  name    = "application-cluster-${local.deployment_id}"
-  region  = local.region
+  name    = "application-cluster-production"
```

Comment:
Removing the possibility of a development step feels like a bad idea. One of the major advantages of our current `puppet` setup for example is the ability to run a full version of the server for yourself and poke it to make changes, without needing to develop in production.

I appreciate that this makes things a little bit simpler, however _especially_ as we're not going to be working with this config very often I think it's important that we have a way to test & validate changes without deploying them.

Reply:
The main complexities came from having an entire duplication of the entire setup. Allowing the provisioning of a development k8s cluster following the same config may be simple enough.

There's no real way to test the changes "locally", but the ability to duplicate the cluster based on an existing config is probably a very useful one!

I'll see what I can do without adding a huge maintenance overhead!

Reply:
I think the correct way to duplicate a cluster if needed is to either do it ad-hoc through the web UI, or duplicate the `resource` and apply.

Note that that only defines that a k8s cluster should exist, rather than what it contains. Duplicating everything wouldn't make sense as it'd need DNS.

Reply:
Hrm, what do we expect the development workflow to look like then?
I think the gold standard here is that there's a single, simple, way to take this configuration and spin up a completely isolated (and ideally fully-local) duplicate copy of it which you can poke manually and/or run automated tests against.
I don't know how much that's even possible for this sort of thing, however I think we should aim to get reasonably close to it if we can.

Importantly, I think we do want it to be possible to go through edit-run-debug cycles without needing actual access to our production infrastructure. That said, I don't know if that's actually possible with our current setup.

Reply:
This specifically configures the DO side of the kubernetes cluster. Everything which would be needed to spin it up locally exists inside the [website repo](https://github.com/srobo/website/tree/master/_env/kubernetes). 

All this code does is create a *blank* cluster in DO. If people want to do that identically in DO, they simply duplicate this block, `terraform apply`, and it'll spin up completely isolated (assuming they change the name, but DO should complain about that).

As an aside, personally I basically consider the k8s cluster deprecated, and hope to move off it "soon", so i'm not concerned with local development needs.